src/simulationarchive.c

```diff
@@ -622,13 +626,31 @@ void reb_simulationarchive_snapshot(struct reb_simulation* const r, const char*
             char* buf_diff;
             size_t size_diff;
             reb_binary_diff(buf_old, size_old, buf_new, size_new, &buf_diff, &size_diff);
-            
-            // Update blob info and Write diff to binary file
+    
             struct reb_simulationarchive_blob blob = {0};
-            fseek(of, -sizeof(struct reb_simulationarchive_blob), SEEK_END);  
+            fseek(of, size_old, SEEK_SET);
+         
+            // Find last valid snapshot to allow for restarting and appending to archives where last snapshot was cut off
+            do
+            {
```

Comment:
I think this loop is problematic. We should probably not entered it by default. Only for corrupt archives. It could be very slow to scan through a few GBs of data everytime a snapshot is appended. Note that in the current implementation we only read in the beginning and the end.